ZenPacks/zenoss/PostgreSQL/poll_postgres.py

```diff
@@ -100,7 +100,14 @@ def getData(self):
 
             dbTableSummaries = copy.copy(tableSummaries)
 
-            databases = pg.getDatabaseStats()
+            # Catch exception in DB query to close open pg connection
+            try:
+                databases = pg.getDatabaseStats()
+            except Exception as ex:
+                print 'Exception in getDatabaseStats(): [{}]'.format(ex)
```

Comment:
This appears to swallow all exceptions, rather than allowing them to be caught by the try block in printJSON, which is what reports them as errors.

It also, by printing a text string, breaks the json parser, which causes all data to be silently discarded, unless I am missing something.

Seems to me that you would want to incorporate this cleanup (pg.close) into that outer try block, so that if any exception occurs, it both raises the event and closes the database connection.